org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/complete/CompletionParser.java

```diff
@@ -5794,13 +5794,27 @@ private MessageSend internalNewMessageSend() {
 }
 @Override
 protected AllocationExpression newAllocationExpression(boolean isQualified) {
-	if (this.assistNode != null && this.lParenPos == this.assistNode.sourceEnd) {
+	if (this.assistNode != null && checkIfAtFirstArgumentOfConstructor()) {
 		CompletionOnQualifiedAllocationExpression allocation = new CompletionOnQualifiedAllocationExpression();
 		this.assistNode = allocation;
 		return allocation;
 	}
 	return super.newAllocationExpression(isQualified);
 }
+private boolean checkIfAtFirstArgumentOfConstructor() {
+	// See if we are in a simple constructure like Foo(|) or Foo(|null, null)
+	if(this.lParenPos == this.assistNode.sourceEnd) {
+		return true;
+	}
+	// We try to handle the situation where lParenPos represents a argument MessageSend's left paren in the AllocationExpression like
+	// Foo(|null, Collections.empty(), null). Here we try to calculate the constructor's left paren and match it with the 
+	// assistNode position.
+	if(this.assistNode instanceof CompletionOnSingleNameReference) {
+		int startPos = this.intStack[this.intPtr] + this.identifierStack[this.identifierPtr].length + 1 /*(*/ + 3 /*new*/;
```

Comment:
I don't understand this unfortunately. The comment here mentions `new` , but the comment above doesn't?

Reply:
The comments are supposed to give a hint about the int literals used for positions calculations. But i will try to make it more clear in a better way.